src/test/java/org/broadinstitute/hellbender/engine/MultiVariantDataSourceUnitTest.java

```diff
@@ -192,6 +192,63 @@ public void testSerialQueries() {
 
     }
 
+    private String getKey(VariantContext vc) {
+        return vc.getContig() + ":" + vc.getStart() + "-" + vc.getEnd();
+    }
+
+    @Test
+    public void testVariantSource() {
+        List<FeatureInput<VariantContext>> featureInputs = new ArrayList<>();
+
+        File vcfPath1 = new File(MULTI_VARIANT_TEST_DIRECTORY, "interleavedVariants_1.vcf");
+        File vcfPath2 = new File(MULTI_VARIANT_TEST_DIRECTORY, "interleavedVariants_2.vcf");
+
+        featureInputs.add(new FeatureInput<>(vcfPath1.getAbsolutePath(), "interleavedVariants_1"));
+        featureInputs.add(new FeatureInput<>(vcfPath2.getAbsolutePath(), "interleavedVariants_2"));
+
+        final Map<String, String> variantToSource = new HashMap<>();
+        try (VCFFileReader reader = new VCFFileReader(vcfPath1)) {
+            reader.iterator().stream().forEach(vc -> {
+                String key = getKey(vc);
+                Assert.assertFalse(variantToSource.containsKey(key));
+                variantToSource.put(key, "interleavedVariants_1");
+            });
+        }
+
+        try (VCFFileReader reader = new VCFFileReader(vcfPath2)) {
+            reader.iterator().stream().forEach(vc -> {
+                String key = getKey(vc);
+                Assert.assertFalse(variantToSource.containsKey(key));
+                variantToSource.put(key, "interleavedVariants_2");
+            });
+        }
+
+        try (final MultiVariantDataSource multiVariantSource =
+                     new MultiVariantDataSource(featureInputs, FeatureDataSource.DEFAULT_QUERY_LOOKAHEAD_BASES)) {
+            int count = 0;
```

Comment:
The variant counts can be removed since they don't contribute anything to what this test is testing.

Reply:
done. i also renamed vcfPath -> vcf since it is no longer actually the path.